src/main/java/org/broadinstitute/hellbender/tools/longreads/graph/AlignedBaseGraphCollection.java

```diff
@@ -0,0 +1,441 @@
+        }
+
+        for ( final Map.Entry<String, AlignedBaseGraph> entry : contigSubGraphMap.entrySet() ) {
+
+            final String contig = entry.getKey();
+            final AlignedBaseGraph graph = entry.getValue();
+
+            final String fileBaseName = baseName + "aligned_base_graph." + contig;
+
+            if ( isGfa2 ) {
+                graph.serializeToGfa2Files(fileBaseName);
+            }
+            else {
+                graph.serializeToGfa1Files(fileBaseName);
+            }
+        }
+    }
+
+    /**
+     * Collapse all linear chains of nodes into one large node containing all bases in the collapsed nodes.
+     * This should be done before writing the graph to GFA format so you can stay sane.
+     */
+    public void collapseAdjacentNodes() {
+
+        // Zip all linear chains:
+        for ( final AlignedBaseGraph graph : contigSubGraphMap.values() ) {
+
+            // First zip chains to make the next step faster.
+            graph.zipLinearChains();
+
+            // Here we must check for adjacent positions in the which have no links between them and add such links.
+            // We should do this because we know they must be adjacent based on the alignment information we were given
+            // in the supporting reads.
+            linkAdjacentNodes(graph);
+
+            // Because we've modified our graphs, we must Zip all linear chains again.
+            // It will be faster this time around because there are fewer nodes in the graph.
+            graph.zipLinearChains();
+
+            //NOTE: If we want to add more nodes after this it will be hard because
+            //      we don't have single base nodes anymore, so the pileups will be hard.
+            //      For now, we do not allow it.
+
+            // Now we rebuild the contigPositionVertexMap for this contig so we can write out the data later:
+            rebuildContigPositionVertexMap(graph);
+        }
+
+        isGraphCollapsed = true;
+    }
+
+    private void rebuildContigPositionVertexMap(final AlignedBaseGraph graph) {
+        final AlignedBaseVertex firstNode = graph.getFirstNode();
+        final String contig = firstNode.getPos().getContig();
+
+        // Clear the existing map:
+        contigPositionVertexMap.get( contig ).clear();
+
+        // Iterate through all nodes in the graph and add them in:
+        for ( final SeqVertex v : graph.vertexSet() ) {
+            final AlignedBaseVertex vertex = (AlignedBaseVertex) v;
+
+            // Add in our holding set (if needed) and any nodes at that position:
+            if ( !contigPositionVertexMap.get( contig ).containsKey(vertex.getPos()) ) {
+                contigPositionVertexMap.get( contig ).put( vertex.getPos(), new HashSet<>() );
+            }
+            contigPositionVertexMap.get( contig ).get( vertex.getPos() ).add( vertex );
+        }
+    }
+
+    private static void linkAdjacentNodes(final AlignedBaseGraph graph) {
```

Comment:
This can happen if you have reads that exactly abut one another.  Because we can resolve the positions in each read to genomic positions, we can know when reads are adjacent.  If reads are exactly adjacent we can merge them.

As of now this is commented out.